content/microservices/basics.md

```diff
@@ -280,6 +280,36 @@ async onApplicationBootstrap() {
 
 If the connection cannot be created, the `connect()` method will reject with the corresponding error object.
 
+#### Extending the ClientProxy
+
+If you need to add some custom logic around the serialization of responses on the client side, you can use a custom class that extends the `ClientProxy` class or one of its child classes. For modifying successful requests you can override the `serializeResponse` method, and for modifying any errors that go through this client you can override the `serializeError` method. To make use of this custom class, you can pass the class itself to the `ClientsModule.register()` method using the `customClass` property. Below is an example of a custom `ClientProxy` that serializes each error into an `RpcException`.
+
+```ts
+@@filename(error-handling.proxy)
+import { ClientTcp, RpcException } from '@nestjs/microservices';
+
+class ErrorHandlingProxy extends ClientTCP {
+  serializeError(err) {
+    return new RpcException(err);
+  }
+}
+```
+
```

Comment:
Can we add something here to separate these 2 code examples? (for example, 1 sentence)

Reply:
Done :)